contracts/JoepegExchange.sol

```diff
@@ -560,6 +594,53 @@ contract JoepegExchange is
         }
     }
 
+    /**
+     * @notice Match multiple asks with their respective taker bid order using AVAX and WAVAX, ignoring expired maker asks
+     * @dev Used when the caller doesn't want the transaction to revert if a maker ask already expired
+     * @param trades an array of trades
+     */
+    function batchBuyWithAVAXAndWAVAXIgnoringExpiredAsks(
+        Trade[] calldata trades
+    ) external payable nonReentrant {
```

Comment:
I don't know, but for the front end it might be useful to return an array of boolean, to know which NFTs were bought?

Reply:
hm yeah good point

Reply:
Not sure how to easily do it

Reply:
(bool return) Return an array of bool here, and change the `_matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks` to return a boolean, I added **(bool return)** on the other functions that would require changes for this snippet to work.
```solidity
    function batchBuyWithAVAXAndWAVAXIgnoringExpiredAsks(
        Trade[] calldata trades
    ) external payable nonReentrant returns (bool[] memory transferStatus /*change name if you have a better one*/) {
        transferStatus = new bool[](trades.length);

        /*
        [...]
        */

        for (uint256 i; i < trades.length; ++i) {
            bool status = _matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks(
                trades[i].takerBid,
                trades[i].makerAsk
            );
           transferStatus[i] = status;
        }
    }
```